src/lib/pg-bark/_impls/alternative-add-appointment.ts

```diff
@@ -0,0 +1,50 @@
+import { Err, Ok, Result } from "@/lib/utilities/result";
+import { CODE } from "@/lib/utilities/bark-code";
+import { PgBarkServiceConfig } from "../pg-bark-service";
+import { dbBegin, dbCommit, dbRelease, dbRollback } from "@/lib/data/db-utils";
+import { selectAppointmentStats } from "../_queries/select-appointment-stats";
+import { insertAppointment } from "../_queries/insert-appointment";
+
+export async function addAppointment(
+  config: PgBarkServiceConfig,
+  args: {
+    dogId: string;
+    vetId: string;
+  },
+): Promise<
+  Result<
+    { appointmentId: string },
+    | typeof CODE.ERROR_DOG_NOT_FOUND
+    | typeof CODE.ERROR_VET_NOT_FOUND
+    | typeof CODE.ERROR_NOT_PREFERRED_VET
+    | typeof CODE.ERROR_APPOINTMENT_ALREADY_EXISTS
+    | typeof CODE.STORAGE_FAILURE
+  >
+> {
+  const conn = await config.dbPool.connect();
+  const { dogId, vetId } = args;
+  try {
+    await dbBegin(conn);
+    const stats = await selectAppointmentStats(conn, { dogId, vetId });
```

Comment:
A better word than "stats" might be "status"; because stats usually refers to statistics, and these are anything but.

Reply:
Renamed to situation.